pkg/reduce/readloop/ordered.go

```diff
@@ -94,15 +100,16 @@ func (of *orderedForwarder) schedulePnF(ctx context.Context,
 // reduceOp invokes the reduce function. The reducer is a long running function since we stream in the data and it has
 // to wait for the close-of-book on the PBQ to materialize the result.
 func (of *orderedForwarder) reduceOp(ctx context.Context, t *task) {
-	// TODO: better logging with partitionID?
 	start := time.Now()
+	defer reduceProcessTime.With(map[string]string{metricspkg.LabelVertex: of.vertexName, metricspkg.LabelPipeline: of.pipelineName}).Observe(time.Since(start).Seconds())
 	for {
 		// FIXME: this error handling won't work with streams. We cannot do infinite retries
 		//  because whatever is written to the stream is lost between retries.
 		err := t.pf.Process(ctx)
 		if err == nil {
 			break
 		} else if err == ctx.Err() {
+			udfError.With(map[string]string{metricspkg.LabelVertex: of.vertexName, metricspkg.LabelPipeline: of.pipelineName}).Inc()
```

Comment:
we may also want to find out if a task spent time waiting to be picked up after the processing is done. for example, a lean task waiting behind a long running task